backends/rapidpro/backend.go

```diff
@@ -336,6 +341,71 @@ func (b *backend) WriteMsgStatus(ctx context.Context, status courier.MsgStatus)
 	return nil
 }
 
+// handleContactURNUpdate handles updating contact URN according to the old URN and the new URN from status
+func (b *backend) handleContactURNUpdate(ctx context.Context, status courier.MsgStatus) error {
+	// both URNs must have the same scheme
+	if status.OldURN().Scheme() != status.NewURN().Scheme() {
+		return errors.New("cannot update contact URN to a different URN scheme")
+	}
+	// ignore update if we have the same URN path
+	if status.OldURN().Path() == status.NewURN().Path() {
+		return nil
+	}
```

Comment:
These checks should be in `SetUpdatedURN` so that any error happens closer to the place where the mistake was made.